sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java

```diff
@@ -1645,6 +1643,44 @@ date_time TIMESTAMP(3) NOT NULL LATENESS INTERVAL 4 SECONDS
         this.statementsFailingInCompilation(sql, "Please use the TABLE function HOP");
     }
 
+    @Test
+    public void issue2529() {
+        String sql = """
+                CREATE TABLE m(
+                   id bigint,
+                   ts timestamp not null lateness interval 1 days,
+                   data int array
+                );
+                create view agg1 as 
+                select max(id)
+                from m
+                group by ts;
+                create view flattened as 
+                select id, v, ts
+                from m, unnest(data) as v;
+                create view agg2 as
+                select max(id)
+                from flattened
+                group by ts;
+                """;
+        CompilerCircuitStream ccs = this.getCCS(sql);
+        this.addRustTestCase(ccs);
+        CircuitVisitor visitor = new CircuitVisitor(new StderrErrorReporter()) {
+            int count = 0;
+
+            @Override
+            public void postorder(DBSPIntegrateTraceRetainKeysOperator operator) {
+                this.count++;
+            }
+
+            @Override
+            public void endVisit() {
+                Assert.assertEquals(4, this.count);
```

Comment:
Do we want to also test that the circuit computes correct results?

Reply:
On which input?

Reply:
Any minimal test input.

Reply:
I will add an I/O test in the next PR